### DISCUSSION TITLE
How to Limit Real-Time Updates to RPC-Fetched Data in Supabase?

### DISCUSSION THREAD

**Discussion by vishwaprojects** [supabase/supabase#31856] [Questions] [answered]
**Title**: How to Limit Real-Time Updates to RPC-Fetched Data in Supabase?  

**Body**:  
I'm using Supabase for a food delivery app and have implemented an RPC function (`nearby_restaurants`) to fetch the nearest 10 restaurants based on the user's location. The function works perfectly and returns the desired data.  

Here's how I fetch the data:  

```dart
getAllRestaurentsRealtime() async {
  try {
    final restaurentsMapList = await _client.rpc('nearby_restaurants', params: {
      'lat': 17.4567743,
      'long': 78.4989977,
    });

    // Parse and store the fetched restaurants
    _restaurentsMap = parseRestaurentMap(restaurentsMapList);
    notifyListeners();
  } catch (e) {
    debugLog('Error calling nearby_restaurants: $e');
  }
}
```

I've also set up a real-time subscription to listen for updates on the `restaurants` table:  

```dart
realtime() async {
  _client
      .channel('restaurants')
      .onPostgresChanges(
        event: PostgresChangeEvent.update,
        schema: 'public',
        table: 'restaurants',
        callback: (payload) {
          final updatedRestaurent =
              RestaurentModel.fromJson(payload.newRecord);
          _restaurentsMap[updatedRestaurent.restaurantId!] = updatedRestaurent;
          notifyListeners();
        },
      )
      .subscribe();
}
```

**Problem**:  
The issue is that this real-time subscription sends updates for **all restaurants** in the database, including those that are not part of the current fetched list of 10 restaurants.  

**Question**:  
How can I limit the real-time updates to only the restaurants returned by my `nearby_restaurants` RPC function? For example, if a restaurant outside the current list gets updated, I don't want it to affect my app state.  

**What I’ve Tried**:  
- Maintaining a list of relevant restaurant IDs (`_relevantRestaurantIds`) from the RPC response and checking against it in the real-time callback.  
- Ignoring updates for IDs not in this list.  

**Is there a better or more efficient way to achieve this using Supabase or database-side filters?**  

Any suggestions or guidance would be greatly appreciated! Thank you!

**Comment by GaryAustin1** [comment, marked as answer]
If you have the list of restaurant id's from the RPC then you can use the "in" filter option on realtime to only return updates for those restaurants.

**Reply by vishwaprojects** [reply]
thank you @GaryAustin1
